### PR TITLE
🔧(releases/*) set replicaset and read_preference in mongodb connection

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

We want to configure mongodb connection using replicaSet and
read_preference parameters. We use default values if nothing is set.

## Proposal

- [x] Add 2 new settings `MONGODB_REPLICASET` and `MONGODB_READ_PREFERENCE`

### CODE REVIEW

File: releases/dogwood/3/bare/CHANGELOG.md [hunk 1]
```diff
@@ -23,4 +27,4 @@ First experimental release of OpenEdx `dogwood.3` (bare flavor).
 
 [unreleased]: https://github.com/openfun/openedx-docker/compare/dogwood.3-1.1.0...HEAD
 [dogwood.3-1.1.0]: https://github.com/openfun/openedx-docker/compare/tag/dogwood.3-1.0.0...dogwood.3-1.1.0
-[dogwood.3-1.0.0]: https://github.com/openfun/openedx-docker/releases/tag/dogwood.3-1.0.0
\ No newline at end of file
+[dogwood.3-1.0.0]: https://github.com/openfun/openedx-docker/releases/tag/dogwood.3-1.0.0
```
Comment (sampaccoud) [inline releases/dogwood/3/bare/CHANGELOG.md]: why this line change :man_shrugging:

File: releases/dogwood/3/bare/config/cms/docker_run_production.py [hunk 2]
```diff
@@ -416,6 +418,8 @@
         "user": MONGODB_USER,
         "password": MONGODB_PASSWORD,
         "ssl": MONGODB_SSL,
+        "replicaSet": MONGODB_REPLICASET,
+        "read_preference": MONGODB_READ_PREFERENCE
```
Comment (sampaccoud) [inline releases/dogwood/3/bare/config/cms/docker_run_production.py]: missing comma for Black... same for all similar files below.

File: releases/dogwood/3/bare/CHANGELOG.md (hunk 1 above)
Comment (lunika) [reply releases/dogwood/3/bare/CHANGELOG.md]: maybe a new line was deleted ?

File: releases/dogwood/3/bare/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- set replicaSet and read_preference in mongodb connection
```
Comment (jmaupetit) [inline releases/dogwood/3/bare/CHANGELOG.md]:
```suggestion
- Set replicaSet and read_preference in mongodb connection
```

File: releases/dogwood/3/fun/CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- set replicaSet and read_preference in mongodb connection
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/CHANGELOG.md]:
```suggestion
- Set replicaSet and read_preference in mongodb connection
```

File: releases/eucalyptus/3/bare/CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- set replicaSet and read_preference in mongodb connection
```
Comment (jmaupetit) [inline releases/eucalyptus/3/bare/CHANGELOG.md]:
```suggestion
- Set replicaSet and read_preference in mongodb connection
```

File: releases/hawthorn/1/bare/CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- set replicaSet and read_preference in mongodb connection
```
Comment (jmaupetit) [inline releases/hawthorn/1/bare/CHANGELOG.md]:
```suggestion
- Set replicaSet and read_preference in mongodb connection
```

File: releases/hawthorn/1/oee/CHANGELOG.md [hunk 7]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- set replicaSet and read_preference in mongodb connection
```
Comment (jmaupetit) [inline releases/hawthorn/1/oee/CHANGELOG.md]:
```suggestion
- Set replicaSet and read_preference in mongodb connection
```

File: releases/dogwood/3/bare/config/cms/docker_run_production.py [hunk 8]
```diff
@@ -405,6 +405,8 @@
 MONGODB_NAME = config("MONGODB_NAME", default="edxapp")
 MONGODB_USER = config("MONGODB_USER", default=None)
 MONGODB_SSL = config("MONGODB_SSL", default=False, formatter=bool)
+MONGODB_REPLICASET = config("MONGODB_REPLICASET", default=None)
+MONGODB_READ_PREFERENCE = config("MONGODB_READ_PREFERENCE", default="PRIMARY")
```
Comment (sampaccoud) [inline releases/dogwood/3/bare/config/cms/docker_run_production.py]: In view of the bug we found out in Dogwood that makes read_preference unusable in this realease, I think you should not include this setting here.

File: releases/dogwood/3/bare/config/lms/docker_run_production.py [hunk 9]
```diff
@@ -628,6 +628,8 @@
 MONGODB_NAME = config("MONGODB_NAME", default="edxapp")
 MONGODB_USER = config("MONGODB_USER", default=None)
 MONGODB_SSL = config("MONGODB_SSL", default=False, formatter=bool)
+MONGODB_REPLICASET = config("MONGODB_REPLICASET", default=None)
+MONGODB_READ_PREFERENCE = config("MONGODB_READ_PREFERENCE", default="PRIMARY")
```
Comment (sampaccoud) [inline releases/dogwood/3/bare/config/lms/docker_run_production.py]: In view of the bug we found out in Dogwood that makes read_preference unusable in this realease, I think you should not include this setting here.

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 10]
```diff
@@ -400,6 +400,8 @@
 MONGODB_NAME = config("MONGODB_NAME", default="edxapp")
 MONGODB_USER = config("MONGODB_USER", default=None)
 MONGODB_SSL = config("MONGODB_SSL", default=False, formatter=bool)
+MONGODB_REPLICASET = config("MONGODB_REPLICASET", default=None)
+MONGODB_READ_PREFERENCE = config("MONGODB_READ_PREFERENCE", default="PRIMARY")
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py]: Add a comment with explanations how to configure this setting? + Maybe a link to pymongo's documentation for explanations on the different modes?
If you do, add it on cms & lms for all releases...

**Comment by lunika** [comment]
> Did you forget the `master/bare` release or is it intended?

Oups I didn't saw this directory 😂

**Pull request merged**